pims/cine.py

```diff
@@ -288,13 +358,52 @@ def __init__(self, filename):
                                                    })
         self.stack_meta_data['trigger_time'] = self.trigger_time
 
+        ### IMAGES
+        # TODO: add support for reading sequence within the same framework, when data
+        # has been saved in another format (.tif, image sequence, etc)
+        if splitext(filename)[-1] in ['.cine', '.cci', '.cin']:
+            self.image_locations = self._unpack('%dQ' % self.image_count,
+                                               self.off_image_offsets)
+            if type(self.image_locations) not in (list, tuple):
+                self.image_locations = [self.image_locations]
+
+    def _clean_setup_dict(self):
+        """Remove obsolete fields and trailing blank characters \x00."""
+        # Filter out 'Res_' (obsolete) fields
+        setup = self.setup_fields_dict
+        k_res = [k for k in setup.keys() if k.startswith('Res_')]
+        for k in k_res:
+            del setup[k]
+        # Remove blank characters
+        self._remove_trailing_x00(setup)
+        # Format f_tone properly
+        tone = setup['f_tone']
+        setup['f_tone'] = tuple((tone[2*k], tone[2*k+1])\
+                                for k in range(setup['tone_points']))
+        return None
+
+    def _remove_trailing_x00(self, dic):
+        for k, v in dic.items():
+            if isinstance(v, bytes):
+                try:
+                    dic[k] = v.decode('utf8').replace('\x00', '')
+                except:
+                    pass
+            elif isinstance(v, Iterable):
+                try:
+                    dic[k] = [el.decode('utf8').replace('\x00', '')\
+                              for el in v]
+                except:
+                    pass
+
     @property
     def filename(self):
         return self._filename
 
     @property
     def frame_rate(self):
-        return self.setup_fields_dict['frame_rate']
+        """Actual frame rate, averaged on frame timestamp (Hz)."""
```

Comment:
I think this is a definite improvement! This is the correct behavior in nearly all circumstances, and it is good to have the `frame_rate` attribute available also.

Reply:
Same remark as above: `frame_rate`, `get_fps`, `setup_fields_dict['frame_rate']` keep the same behaviour. `frame_rate_avg` and `_compute_frame_rate()` are introduced.